File: src/main/java/com/googongill/aditory/external/s3/dto/S3DownloadResult.java
package com.googongill.aditory.external.s3.dto;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class S3DownloadResult {
    private Long profileImageId;
    private String originalName;
    private String url;

    public static S3DownloadResult of(Long profileImageId, String originalName, String url) {
        return S3DownloadResult.builder()
                .profileImageId(profileImageId)
                .originalName(originalName)
                .url(url)
                .build();
    }
}

File: src/main/java/com/googongill/aditory/external/s3/dto/S3DownloadResult.kt
package com.googongill.aditory.external.s3.dto

data class S3DownloadResult(
    val profileImageId: Long,
    val originalName: String,
    val url: String
) {
    companion object {
        fun of(profileImageId: Long, originalName: String, url: String): S3DownloadResult {
            return S3DownloadResult(
                profileImageId = profileImageId,
                originalName = originalName,
                url = url
            )
        }
    }

}
